module/entities/helper.js

```diff
@@ -1529,57 +1528,44 @@ module.exports = class EntitiesHelper {
                   return resolve(response);
 
               } else {
-                let groups = entitiesDetails.data[0].groups;
-
-                let targetedGroup = groups[result.entityType];
-                let filterDataGroups = {
-                  "_id":targetedGroup,
-                  "tenantId":req.userDetails.tenantData.tenantId,
-                  "orgIds": {$in:['ALL',req.userDetails.tenantData.orgId]}
-                }
-                let projections = ['entityType','metaInformation.externalId', 'metaInformation.name']
+              // Fetch data from entity service
 
-                entitiesDetails = await entityManagementService.entityDocuments(filterDataGroups,projections,req.pageNo,req.pageSize,req.searchText);
-                if ( !entitiesDetails.success ) {
-                  return resolve({
-                      "message" : messageConstants.apiResponses.ENTITY_NOT_FOUND,
-                      "result" : [{
-                          "count":0,
-                          "data" : []
-                      }]
-                  })
+              let projections = ['_id','entityType','metaInformation.externalId', 'metaInformation.name']
+              entitiesDetails = await entityManagementService.entityDocuments(filterData,[],req.pageNo,req.pageSize,req.searchText,`$groups.${result.entityType}`, true, false, projections);
```

Comment:
@Prajwal17Tunerlabs can we add comments to understand the params , add some descriptions